provider/azure/terraform/example/helpers.rb

```diff
@@ -40,8 +40,9 @@ def flatten_example_properties_to_check(properties, has_nested_item)
             has_nested_item = false
             properties.each do |pn, pv|
               if pv.is_a?(Hash)
+                flat_properties["#{pn}.#"] = pv.empty? ? 0 : 1
                 pv.each do |key, val|
-                  flat_properties["#{pn}.#{key}"] = val
+                  flat_properties["#{pn}.0.#{key}"] = val
```

Comment:
For hash items, this number should be the hash value of that item instead of just `0`. Typically terraform uses [these functions](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/helpers/set/set.go) to calculate hash codes, which will eventually calls into [`hashcode.String`](https://github.com/hashicorp/terraform/blob/master/helper/hashcode/hashcode.go#L14). So it might not be a simple job to mimic this Go code in Ruby.